src/V3Assert.cpp

```diff
@@ -651,16 +653,29 @@ class AssertVisitor final : public VNVisitor {
         m_inSampled = false;
         iterateChildren(nodep);
     }
-    void visit(AstPExpr* nodep) override {
-        {
-            VL_RESTORER(m_inSampled);
-            m_inSampled = false;
-            iterateAndNextNull(nodep->precondp());
+    void visit(AstPExprClause* nodep) override {
+        if (m_underAssert) {
+            if (nodep->passs() && m_passsp) {
+                // Cover adds COVERINC by AstNode::addNext, thus need to clone next too.
+                nodep->replaceWith(m_passsp->cloneTree(true));
+                VL_DO_DANGLING(pushDeletep(nodep), nodep);
+            } else if (!nodep->passs() && m_failsp) {
+                // Asserts with multiple statements are wrapped in implicit begin/end blocks so no
+                // need to clone next.
+                nodep->replaceWith(m_failsp->cloneTree(false));
+                VL_DO_DANGLING(pushDeletep(nodep), nodep);
+            } else {
+                VL_DO_DANGLING(pushDeletep(nodep->unlinkFrBack()), nodep);
```

Comment:
The delete is in all 3 branches so move below, and put just a nodep->unlinkFrBack() here.